src/hb-ot-layout.cc

```diff
@@ -318,6 +318,37 @@ hb_ot_layout_get_glyphs_in_class (hb_face_t                  *face,
   return face->table.GDEF->table->get_glyphs_in_class (klass, glyphs);
 }
 
+#ifndef HB_NO_VAR
+/**
+ * hb_ot_layout_collect_variation_indices:
+ * @face: The #hb_face_t to work on
+ * @layout_variation_indices: (inout): The #hb_set_t set of var indexes in all Device
+ * tables
+ *
+ * Fetch and remap a list of variation indices in all Device tables referenced
+ * in the specified face's GDEF table and GPOS tables that are going to be
+ * retained in the final subset
+ *
+ * Since: REPLACEME
+ **/
+void
+hb_ot_layout_collect_variation_indices (hb_face_t      *face,
+					const hb_set_t *glyphset,
+					const hb_map_t *gpos_lookups,
+					hb_set_t       *layout_variation_indices /* OUT */,
+					hb_map_t       *layout_variation_idx_map /* OUT */)
```

Comment:
Let's avoid making this a new public API method since it's only useful within subset plan creation. I threw together a quick commit which shows how this could be moved into subset plan: https://github.com/googlefonts/harfbuzz/pull/new/GDEF-VariationStore-subset_no_public_api

Reply:
Done. thanks!